src/platform/testing/e2e/cypress/plugins/index.js

```diff
@@ -12,60 +11,75 @@ const tableConfig = {
 };
 
 module.exports = async on => {
-  const ENV = 'localhost';
-  const publicPath = '/generated/';
-  let outputOptions = {};
+  let appRegistry;
+  if (process.env.CYPRESS_CI) {
+    // eslint-disable-next-line import/no-unresolved
+    appRegistry = require('../../../../../../content-build/src/applications/registry.json');
+  } else {
+    appRegistry = require('../../../../../../../content-build/src/applications/registry.json');
+  }
+  // eslint-disable-next-line no-useless-escape
+  const cypressPlugin = {
+    name: 'cypress',
 
-  // Import our own Webpack config.
-  await require('../../../../../../config/webpack.config.js')(ENV).then(
-    webpackConfig => {
-      // Get the original DefinePlugin so we can use __REGISTRY__
-      const definePlugin = webpackConfig.plugins.find(
-        plugin => plugin.constructor.name === 'DefinePlugin',
-      );
-
-      const appRegistry = definePlugin.definitions.__REGISTRY__;
-
-      const options = {
-        webpackOptions: {
-          ...webpackConfig,
-          plugins: [
-            new DefinePlugin({
-              __BUILDTYPE__: JSON.stringify(ENV),
-              __API__: JSON.stringify(''),
-              __REGISTRY__: appRegistry,
-            }),
-            new ProvidePlugin({
-              process: 'process/browser',
-            }),
-          ],
-
-          // Expose some Node globals.
-          node: {
-            __dirname: true,
-            __filename: true,
-          },
-        },
-      };
-
-      // Webpack 5 workaround to keep Cypress from unsetting publicPath
-      // https://github.com/cypress-io/cypress/issues/8900
-      Object.defineProperty(options.webpackOptions, 'output', {
-        get: () => {
-          return { ...outputOptions, publicPath };
-        },
-        set: x => {
-          outputOptions = x;
-        },
+    setup(build) {
+      // eslint-disable-next-line consistent-return
+      build.onLoad({ filter: /.js?$/ }, ({ path: filePath }) => {
+        if (!filePath.split(path.sep).includes('node_modules')) {
```

Comment:
Would be nice to extract helper functions for complex logic like this to make this clear:

```js
function isDirectoryInPath(directory, filePath) {
  return filePath.split(path.sep).includes(directory);
}